### PR TITLE
[WIP] 🏃 Switch to using go modules #2127

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Fixes: #1886

Signed-off-by: Ashish Amarnath ashisham@vmware.com

### CODE REVIEW

File: pkg/plugin/clientmgmt/logrus_adapter.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const pluginNameField = "pluginName"
 // delegates all calls to a logrus logger.
 type logrusAdapter struct {
 	impl  logrus.FieldLogger
-	level logrus.Level
+	level hclog.Level
```
Comment (nrb) [inline pkg/plugin/clientmgmt/logrus_adapter.go]: Why replace logrus as part of this?

Comment (ashish-amarnath) [reply pkg/plugin/clientmgmt/logrus_adapter.go]: On switching to using go modules, I could not get `hclog` library version w/o the `SetLevel` method  in the interface definition. Now, to use our logrus adapter as a hclog logger I had to implement 
```
func (l *logrusAdapter) SetLevel(level hclog.Level) {
	l.level = level
}
```
Now as I type this, I think I can I can do something like
```
hclog.LevelFromString(logLevel.String())
```

here to keep the changes minimal.

Comment (ashish-amarnath) [reply pkg/plugin/clientmgmt/logrus_adapter.go]: I take that back. There is no way to convert a `hclog.Level` to `logrus.Level` even though they are are both `int32` and `uint32`.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	k8s.io/apiextensions-apiserver v0.0.0
+	k8s.io/apimachinery v0.0.0
+	k8s.io/client-go v0.0.0
+	k8s.io/code-generator v0.17.0 // indirect
+	k8s.io/klog v0.3.1
+	k8s.io/kubernetes v1.15.3
+	sigs.k8s.io/controller-tools v0.2.4 // indirect
+)
+
+replace (
```
Comment (skriss) [inline go.mod]: how did you come up with these `replace` directives/why are the needed?

Comment (ashish-amarnath) [reply go.mod]: These were needed to resolve errors w/ the controller-gen code generation. The way i came up w/ these was a combination of searching on the internet on how to resolve those errors and https://github.com/vmware-tanzu/velero/blob/master/Gopkg.lock#L671 replaces corresponding to the 1.15.3

**Comment by carlisia** [comment]
@ashish-amarnath I think this PR is obsolete in lieu of https://github.com/vmware-tanzu/velero/pull/2214, right?

**Comment by skriss** [comment]
will confirm with @ashish-amarnath but yeah, I think we can close this one out.

**Comment by ashish-amarnath** [comment]
Yes, this is obsolete PR. I am using this to review PR #2214 
Closing this PR

**Pull request closed without merging**